runtime/compilers/rillv1/parse_dotenv.go

```diff
@@ -2,25 +2,50 @@ package rillv1
 
 import (
 	"context"
+	"fmt"
+	"maps"
 	"os"
 
 	"github.com/joho/godotenv"
 )
 
-// parseDotEnv parses the env file present at repo root
-func (p *Parser) parseDotEnv(ctx context.Context, path string) error {
+// parseAndMergeEnvironmentVariables parses the .env file at the given path and merges it with the existing env vars
+func (p *Parser) parseAndMergeEnvironmentVariables(ctx context.Context, path string) error {
```

Comment:
Let's keep the `parseDotEnv` name, all the file-type-specific parse functions in the package have a name like `parseXYZ` (and they all internally modify the parser's state).